### PR TITLE
Update Bar unit tests to use Jest

### PR DESCRIPTION
**Pull request by sdadn**
### Summary

This PR updates the repo to utilize Jest for unit tests instead of Karma. Since there are a large amount of tests to go over, this PR narrows the scope to only the Bar graph. Other graph types and unit tests will be updated in subsequent PRs. 

**Note:** A small subset of tests are failing due to differing values or due to an undefined value in the graph API. These tests are being skipped for now as they require dedicated debugging and the current objective is to switch over to Jest for unit tests. These failing tests will be fixed in a separate Jira. 

### Testing

This change was tested using:

- [ ] WDIO
- [x] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

The CICD was switched over to utilize Jest and it is currently passing:

CleanShot 2024-01-08 at 13 25 39

CleanShot 2024-01-08 at 13 25 47

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9813 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/carbon-graphs/tests/unit/controls/Bar/AxisInfoRow.test.js [hunk 1]
```diff
@@ -1,6 +1,5 @@
-'use strict';
+// 'use strict';
```
Comment (adoroshk) [inline packages/carbon-graphs/tests/unit/controls/Bar/AxisInfoRow.test.js:1]: Should we remove it?

**Review by adoroshk — APPROVED** [review]

File: packages/carbon-graphs/tests/unit/controls/Bar/AxisInfoRow.test.js (hunk 1 above)
Comment (sdadn) [reply packages/carbon-graphs/tests/unit/controls/Bar/AxisInfoRow.test.js]: There's too many to change at this time so I'm going to revert the comment and keep it for now. It won't hurt to have atm.

File: packages/carbon-graphs/tests/unit/controls/Bar/BarRegions.test.js [hunk 2]
```diff
@@ -0,0 +1,567 @@
+'use strict';
```
[the remark this answers is not in the document]
Comment (sdadn) [reply packages/carbon-graphs/tests/unit/controls/Bar/BarRegions.test.js:1]: We can keep this for now.

**Review by adoroshk — APPROVED** [review]

File: packages/carbon-graphs/babel.config.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  api.cache(false);
+  api.assertVersion('^7.4.4');
+
+  const presets = [
+    '@babel/preset-env',
+    '@babel/preset-react',
+  ];
+
+  const plugins = [
+    '@babel/plugin-proposal-object-rest-spread',
```
[the remark this answers is not in the document]
Comment (sdadn) [reply packages/carbon-graphs/babel.config.js:11]: Thanks for pointing that out! I'll make that change in the next PR.

File: packages/carbon-graphs/tests/unit/controls/Bar/BarPanning.test.js [hunk 4]
```diff
@@ -87,7 +87,7 @@ describe('Bar - Panning', () => {
 
       expect(() => { graphDefault.reflow(panData); }).toThrowError(errors.THROW_MSG_INVALID_DATA);
     });
-    it('throws error when undefined value is passed as y', () => {
+    it('throws aan error when y is passed as undefined', () => {
```
[the remark this answers is not in the document]
Comment (sdadn) [reply packages/carbon-graphs/tests/unit/controls/Bar/BarPanning.test.js]: I believe this got updated in: https://github.com/cerner/terra-graphs/pull/324/commits/6cf7494fc08408cba8bf6690803eb341489e37bb

**Pull request merged**